src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -809,12 +842,22 @@ public ICraftingLink submitJob(final IGrid g, final ICraftingJob job, final Base
         try {
             this.waitingFor.resetStatus();
             job.startCrafting(ci, this, src);
+
+            // Clear the follow list by default
+            this.followPlayerNameList.clear();
+
+            // If need add player into follow list when submit crafting, use the following code
+            // be careful ClassCastException, because machine can submit crafting job too.
+
+            // this.followPlayerNameList.add(((PlayerSource) src).player.getDisplayName());
+
             if (ci.commit(src)) {
                 if (job.getOutput() != null) {
                     this.finalOutput = job.getOutput();
                     this.waiting = false;
                     this.isComplete = false;
                     this.usedStorage = job.getByteTotal();
+                    this.targetItemCount = job.getOutput().getStackSize();
```

Comment:
sry for my bad english, because the origin `startItemCount` will add all sub crafting job's item number.
for example: A + B = C, C + B = D. Now i have 9 C and try to order 10 D, the `startItemCount` is 10 for crafting D and 1 for crafting C, so is 11 total.
to prevent this i make a new var to store this, maybe this name is a bit of misleading, what about change it to `targetItemNumber`?

Reply:
or `numsOfOutput` is better?

Reply:
@MCTBL The name of this field is clear, so there's no need to rename it. I asked for its *purpose*, and now I've got it.

I ask you for 2 changes:
1) First change of this counter (with `+=`) should be moved inside `mergeJob` method. And this change should occur only if merge is successful (so, put it around [here](https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/blob/master/src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java#L908)).
2) Currently this counter is never reset to 0 (and that potentially can lead to incorrect value). Please, reset value of this field. The best place for doing that will be at the end of the method `completeJob`.